### ISSUE TITLE
Uncalled function in binary expression not silenced by body usage

### ISSUE THREAD

**Issue by andrewbranch** [microsoft/TypeScript#49155]
Found in #49142:

```ts
declare let isActive: boolean;
declare let onChange: () => void;

if (!isActive && onChange) onChange();
```

Expected: no error because `onChange()` is used in the conditional body.
Actual: always-truthy error in the `if` condition.
